packages/mask/src/plugins/FileService/SNSAdaptor/Preview.tsx

```diff
@@ -56,7 +56,17 @@ export function Preview({ info }: { info: FileInfo }) {
             {t('plugin_file_service_unencrypted')}
         </Typography>
     )
-    const link = `https://arweave.net/${info.landingTxID}`
+
+    let linkPrefix = null
+    if (info.provider == 'arweave') {
+        linkPrefix = 'https://arweave.net/';
+    } else if (info.provider == 'ipfs') {
+        linkPrefix = 'https://infura-ipfs.io/ipfs/';
+    } else if (info.provider == 'swarm') {
+        linkPrefix = 'https://bee-2.gateway.ethswarm.org/bzz/';
+    }
```

Comment:
Move this to a pipe file and use `unreachable` to refactor it.

Learn more: packages/mask/src/plugins/Trader/pipes.ts

Reply:
plz extract to function.

```ts
function getGatewayAPI(provider: Provider) {
    if (provider == 'arweave') {
        return 'https://arweave.net/';
    } else if (provider == 'ipfs') {
        return 'https://infura-ipfs.io/ipfs/';
    } else if (provider == 'swarm') {
        return 'https://bee-2.gateway.ethswarm.org/bzz/';
    }
    throw new Error('unsupported provider')
}
```